src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -85,6 +97,10 @@ public void run() {
       try {
         runnable.run();
       } finally {
+        if (_pendingCount.get() == 1) {
+          _active = false;
+          _activityHandler.deactivated();
+        }
```

Comment:
This is not thread-safe, right? Someone can increment the _pendingCount after this block and have you tryExecuteLoop() again.